File: modules/java/main/src/main/java/com/mooltiverse/oss/nyx/services/gitlab/GitLabUser.java
/*
 * Copyright 2020 Mooltiverse
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.mooltiverse.oss.nyx.services.gitlab;

import java.util.Map;

import com.mooltiverse.oss.nyx.services.User;

/**
 * A user for a remote GitLab service.
 */
public class GitLabUser extends GitLabEntity implements User {
    /**
     * Creates the user object modelled by the given attributes.
     * 
     * @param api the reference to the API used to communicate with the remote end. Can't be {@code null}
     * @param attributes the map of attributes for this object. Can't be {@code null}
     * 
     * @throws NullPointerException if the given attributes map is {@code null}
     * @throws IllegalArgumentException if the map of attributes is empty
     */
     GitLabUser(API api, Map<String, Object> attributes) {
        super(api, attributes);
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String getID() {
        return getAttributes().get("id").toString();
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String getUserName() {
        return getAttributes().get("username").toString();
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public String geFullName() {
        // this is not a mandatory attribute so we need to check for nulls
        Object res = getAttributes().get("name");
        return res == null ? null : res.toString();
    }
}

File: src/go/nyx/services/gitlab/gitlab_user.go
/*
 * Copyright 2020 Mooltiverse
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package gitlab

import (
	"strconv" // https://pkg.go.dev/strconv

	gl "gitlab.com/gitlab-org/api/client-go" // https://pkg.go.dev/gitlab.com/gitlab-org/api/client-go
)

/*
A user for a remote GitLab service.
*/
type GitLabUser struct {
	// The ID for the user on this service.
	id string

	// The name for the user on this service.
	userName string

	// The full name for the user on this service.
	fullName string
}

/*
Creates the user object modelled by the attributed from the given reference.

Arguments are as follows:

  - user the object to read the attributes from
*/
func newGitLabUser(user gl.User) *GitLabUser {
	res := &GitLabUser{}
	res.id = strconv.FormatInt(user.ID, 10)
	res.userName = user.Username
	res.fullName = user.Name
	return res
}

/*
Returns the ID for the user on this service.
*/
func (u *GitLabUser) GetID() string {
	return u.id
}

/*
Returns the name for the user on this service.
*/
func (u *GitLabUser) GetUserName() string {
	return u.userName
}

/*
Returns the full name for the user on this service.
*/
func (u *GitLabUser) GetFullName() string {
	return u.fullName
}
